File: from-0101-to-0200/0198-house-robber/solution.go
func rob(nums []int) int {
    n := len(nums)
    dp := make([]int, n)
    var to_rob, not_to_rob int

    for i := 0; i < n; i ++ {
        to_rob = nums[i]
        if i > 1 {
            to_rob += dp[i - 2]
        }

        not_to_rob = 0
        if i > 0 {
            not_to_rob = dp[i - 1]
        }
        if not_to_rob > to_rob {
            dp[i] = not_to_rob
        } else {
            dp[i] = to_rob
        }
    }
    return dp[n - 1]
}

File: from-0101-to-0200/0198-house-robber/solution.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        dp = [0] * len(nums)
        
        for idx in range(len(nums)):
            to_rob = (dp[idx-2] + nums[idx]) if idx > 1 else nums[idx]
            not_to_rob = dp[idx-1] if idx > 0 else 0
            
            dp[idx] = max(to_rob, not_to_rob)
            
            
        return dp[-1]